File: 2015/day02/part1.py
ans = 0
with open("input.txt") as file:
    for line in file:
        l, w, h = map(int, line.split("x"))
        paper = 2 * l * w + 2 * w * h + 2 * h * l
        area = min(l * w, w * h, h * l)
        ans += paper + area
print(ans)

File: 2025/day02/part1.cc
#include <cstdio>
#include <fstream>
#include <iostream>
#include <sstream>
#include <vector>
using namespace std;

int main() {

  ifstream file("input.txt");
  string line;
  getline(file, line);
  file.close();

  vector<string> ids;

  stringstream ss(line);
  string range;
  while (getline(ss, range, ',')) {
    long long first = 0;
    long long second = 0;
    sscanf(range.c_str(), "%lld-%lld", &first, &second);
    while (first <= second) {
      ids.emplace_back(to_string(first));
      first++;
    }
  }

  long long ans = 0;
  for (string s : ids) {
    string first = s.substr(0, s.length() / 2);
    string second = s.substr(s.length() / 2);
    if (first == second) {
      long long id = stoll(s);
      ans += id;
    }
  }
  cout << ans;

  return 0;
}
